### PR TITLE
Added Basic Testcases 

### LINKED ISSUE #21: Adding basic test suite
We should properly mock the query to ElasticSearch to have the tests work in Github Actions. Please refer to [IntelOwl](https://github.com/intelowlproject/IntelOwl) to get an example

### PR DESCRIPTION
**Pull request by uzaxirr**
# Description

Added Testcases for Views and Models

## Related issues

Fixes #21 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

# Checklist

- [ ] I have read and understood the rules about [how to Contribute](https://intelowl.readthedocs.io/en/latest/Contribute.html) to this project
- [ ] The pull request is for the branch `dev`
- [ ] The tests gave 0 errors.
- [ ] Linters (`Black`, `Flake`, `Isort`) gave 0 errors. If you have correctly installed [pre-commit](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [ ] The commits were squashed into a single one (optional, they will be squashed anyway by the maintainer)
  
### Important Rules
- If your changes decrease the overall tests coverage (you will know after the Codecov CI job is done), you should add the required tests to fix the problem
- Everytime you make changes to the PR and you think the work is done, you should explicitly ask for a review

### CODE REVIEW

**Comment by uzaxirr** [comment]
> First batch of tests makes sense,
> 
> we also need to add those tests in the CI (`pull_request_automation.yml`). See IntelOwl for reference


From what i can figure out is that intelOwl has tests running through a shell script inside a docker container. Is this what you want? I think #23 will fail the docker build.
Do i have to mock ElasticSearch connection for docker build in order to make them pass through?

**Comment by mlodic** [comment]
yes, as already mentioned in the issue https://github.com/honeynet/GreedyBear/issues/21. This could also be a thing for another PR if you want to split the work. The important thing here is to run in the CI at least the new tests that you have created

**Comment by uzaxirr** [comment]
> yes, as already mentioned in the issue #21. This could also be a thing for another PR if you want to split the work. The important thing here is to run in the CI at least the new tests that you have created

Cool then you can merge this PR and i'll create a new for automating it with CI

**Comment by mlodic** [comment]
I meant to have the CI in this PR. Otherwise how can we test if the tests work?

**Comment by uzaxirr** [comment]
@mlodic 
i’m kind of stucked. I successfully got the containers running but the workflow gets stucked in `greedybear_uwsgi | [uWSGI] getting INI configuration from /etc/uwsgi/sites/greedybear.ini`
can you please review it and point out what wrong i’m doing.
For the time being i’ve commented out the ngnix and trying to get the django container up without it. will include ngnix once django’s container works fine

**Comment by mlodic** [comment]
use `-d` so it goes on the background :P

**Comment by uzaxirr** [comment]
> use `-d` so it goes on the background :P

Ohh crap how did i forgot to run it in detach mode ;(*
It worked fine after that.

But now while adding up ngnix i'm facing another roadblock.
```
ERROR: for greedybear_uwsgi  Cannot start service uwsgi: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "./docker/entrypoint_uwsgi.sh": stat ./docker/entrypoint_uwsgi.sh: no such file or directory: unknown

ERROR: for uwsgi  Cannot start service uwsgi: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "./docker/entrypoint_uwsgi.sh": stat ./docker/entrypoint_uwsgi.sh: no such file or directory: unknown
Encountered errors while bringing up the project.
```
It says that `docker/entrypoint_uwsgi.sh` does not exist but it does. i'm unable to understand this. can you please point out the mistake. Sorry in advance if its a silly mistake like the previous one

File: .github/workflows/pull_request_automation.yml [hunk 1]
```diff
@@ -31,4 +31,20 @@ jobs:
...
+      run: |
+        cp .env_template .env
+        cp docker/env_file_template docker/env_file
+        cp docker/env_file_postgres_template docker/env_file_postgres
+    - name: Build and run containers
+      run: |
+        docker-compose -f docker/default.yml -f docker/ci.override.yml up --build -d
+    - name: "Install requirements"
+      run: |
+        sudo docker exec greedybear_uwsgi pip3 install --no-cache-dir --compile -r test-requirements.txt
```
Comment (mlodic) [inline .github/workflows/pull_request_automation.yml]: I guess sudo is not required

File: .github/workflows/pull_request_automation.yml [hunk 2]
```diff
@@ -31,4 +31,20 @@ jobs:
 
     - name: isort
       run: |
-        isort . --profile black --filter-files --check-only --diff --skip venv
\ No newline at end of file
+        isort . --profile black --filter-files --check-only --diff --skip venv
+        
+    - name: Before Build
```
Comment (mlodic) [inline .github/workflows/pull_request_automation.yml:45]: please see how CI is actually done in IntelOwl. Please split linters with the tests

**Comment by mlodic** [comment]
> > use `-d` so it goes on the background :P
> 
> Ohh crap how did i forgot to run it in detach mode ;(* It worked fine after that.
> 
> But now while adding up ngnix i'm facing another roadblock.
> 
> ```
> ERROR: for greedybear_uwsgi  Cannot start service uwsgi: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "./docker/entrypoint_uwsgi.sh": stat ./docker/entrypoint_uwsgi.sh: no such file or directory: unknown
> 
> ERROR: for uwsgi  Cannot start service uwsgi: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "./docker/entrypoint_uwsgi.sh": stat ./docker/entrypoint_uwsgi.sh: no such file or directory: unknown
> Encountered errors while bringing up the project.
> ```
> 
> It says that `docker/entrypoint_uwsgi.sh` does not exist but it does. i'm unable to understand this. can you please point out the mistake. Sorry in advance if its a silly mistake like the previous one

try to make some tests locally with the same commands to get which can be the error

File: .github/workflows/pull_request_automation.yml (hunk 1 above)
Comment (uzaxirr) [reply .github/workflows/pull_request_automation.yml]: Done

File: .github/workflows/pull_request_automation.yml (hunk 2 above)
Comment (uzaxirr) [reply .github/workflows/pull_request_automation.yml:45]: Done

**Comment by uzaxirr** [comment]
Hey @mlodic 
I've been trying to resolve the above mentioned issue since yesterday. Ran it locally and got the same thing tried a lot of things for my end as well as from internet as well but all in vain. Can you please review it.
It worked fine without ngnix (see [this](https://github.com/honeynet/GreedyBear/actions/runs/1950841411) run) but when i introduced ngnix that error keeps on arising. I tallied my implementation with the one in  intelOwl and couldn't find any difference.

Thanks!

File: docker/ci.override.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      dockerfile: docker/Dockerfile
+    image: honeynet/greedybear:ci
+    env_file:
+      - env_file_app_ci
+  
+  nginx:
+    build:
+      context: ..
+      dockerfile: docker/Dockerfile_nginx
+    image: honeynet/greedybear:ci
```
Comment (mlodic) [inline docker/ci.override.yml]: this name should be different from the other one...the dockerfiles are different

Comment (mlodic) [reply docker/ci.override.yml]: call it `honeynet/greedybear_nginx:ci`

Comment (uzaxirr) [reply docker/ci.override.yml]: Thanks a lot for help!!

**Comment by uzaxirr** [comment]
Seems like it is good to go

**Comment by mlodic** [comment]
great! we are done!

**Pull request merged**